src/client_side_reply.cc

```diff
@@ -1410,6 +1410,7 @@ clientReplyContext::buildReplyHeader()
     }
 
     reply->header.removeHopByHopEntries();
+    reply->removeIrrelevantContentLength();
```

Comment:
I would add a comment to both callers: `// paranoid: ContentLengthInterpreter has cleaned non-generated replies`.